extensions/guidepost/common/client.mjs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+import {MapCoord} from '/static/common/maplib/map.mjs';
+
+const LAYER_GUIDEPOST = {zIndex: -10, layerName: "guidepost"};
+/**
+ * This class is the browser/client side of an extension.
+ * One instance is created for each connected player.
+ */
+class Client {
+  /**
+   * Create the client side of the extension.
+   * @constructor
+   * @param {ClientExtensionHelper} helper - An extension helper object for
+   * servicing various functionalities of the extension.
+   */
+  constructor(helper) {
+    this.helper = helper;
+  }
+
+  /**
+   * Initialize the layer and send the guidepost data as a parameter from json file
+   */
+  async gameStart(){
+    const guideposts = game.assetData = await $.ajax({url: "/static/run/map/watermark/guideposts.json", type: "GET",});
+    guideposts.forEach((guidepost, index, guideposts) => {
+      const image = new Image();
+      image.assetName = guidepost.assetName;
+      image.src = guidepost.src;
+      guideposts[index].image = image;  
+    });
```

Comment:
Refer to the following code to prevent any race condition:
https://github.com/canyugs/hitcon-online/blob/0524c346200a147e308e6a80cb3f89d2b1d09cd6/common/maplib/graphic-asset.mjs#L28-L45